internal/k8s/client.go

```diff
@@ -654,24 +654,31 @@ func (c *Client) GetRunningCiliumVersion(ctx context.Context, namespace string)
 		return "", fmt.Errorf("unable to list cilium pods: %w", err)
 	}
 	if len(pods.Items) > 0 && len(pods.Items[0].Spec.Containers) > 0 {
-		image := pods.Items[0].Spec.Containers[0].Image
-		version := strings.SplitN(image, ":", 2)
-		if len(version) != 2 {
-			return "", errors.New("unable to extract cilium version from container image")
-		}
-		v := version[1]
-		if digest := strings.Index(v, "@"); digest > 0 {
-			v = v[:digest]
-		}
-		// Add any part in the pod image separated by a '-` to the version,
-		// e.g., "quay.io/cilium/cilium-ci:1234" -> "-ci:1234"
-		base := strings.Split(version[0], "/")
-		last := base[len(base)-1]
-		dash := strings.Index(last, "-")
-		if dash >= 0 {
-			v = last[dash:] + ":" + v
+		for _, container := range pods.Items[0].Spec.Containers {
+			image := strings.SplitN(container.Image, ":", 2)
+			if len(image) != 2 {
+				// skip invalid images
+				continue
+			}
+			base := strings.Split(image[0], "/")
+			last := base[len(base)-1]
+			if !strings.Contains(last, "cilium") {
```

Comment:
nit: strings.HasPrefix will be stricter validation

Reply:
Updated, please take a look, thanks!